test/lib/commonkafka.py

```diff
@@ -108,80 +96,100 @@ def get_kafka_connector_tasks(setup, params):
     return 0
 
 
-def get_kafka_connector_status(setup, params):
+def get_kafka_connector_status(setup, params, action, state):
     '''
     Get kafka connect connector tasks using kafka connect REST API
     '''
     t_end = time.time() + 10
     while time.time() < t_end:
         response = requests.get(url=setup["kafka_connect_url"] + "/connectors/" + params["name"] + "/status",
                                 headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
-        status = response.status_code
-        if status == 200:
-            return response.json()
+        content = response.json()
+        if content.get('connector'):
+            if content['connector']['state'] == state:
+                return True
+
+    logger.error("Failed to {} connector and tasks are not in a {} state after 10 seconds".format(action, state))
+    return False
+
+
+def get_running_kafka_connector_task_status(setup, params):
+    '''
+    Get running kafka connect connector tasks status using kafka connect REST API
+    '''
+    time.sleep(2)
+    response = requests.get(url=setup["kafka_connect_url"] + "/connectors/" + params["name"] + "/status",
+                            headers={'Accept': 'application/json', 'Content-Type': 'application/json'})
+    content = response.json()
 
-    return None
+    if content.get('connector'):
+        if content['connector']['state'] == 'RUNNING':
+            task_status = jsonpath.jsonpath(content, '$.tasks.*.state')
```

Comment:
if there are multiple tasks, what does it return ?

Reply:
Yes, if I set `"tasks.max": "3"` there will be 3 tasks. I also have test cases to cover that to make sure all the tasks state are 'FAILED'